### PR TITLE
Add ip connection limit option

### LINKED ISSUE #106: Avoid one IP to take all the connections of the pool
libmicrohttps proposes To avoid one MHD_OPTION_PER_IP_CONNECTION_LIMIT which seems to be adapted to that.

### PR DESCRIPTION
**Pull request by juuz0**
Fixes #106
Part 2: https://github.com/kiwix/libkiwix/pull/700

### CODE REVIEW

File: src/server/kiwix-serve.cpp [hunk 1]
```diff
@@ -74,6 +74,7 @@ void usage()
             << "\t-V, --version\t\tprint software version" << std::endl
             << "\t-z, --nodatealiases\tcreate URL aliases for each content by removing the date" << std::endl
             << "\t-c, --customIndex\tadd path to custom index.html for welcome page" << std::endl
+            << "\t-L, --ipConnectionLimit\tLimit on the number of (concurrent) connections made to the server from the same IP address (default: infinite, recommended: >= 6)" << std::endl
```
Comment (veloman-yunkan) [inline src/server/kiwix-serve.cpp:77]: Unlike other options the description of this new option starts with an uppercase letter.

**Review by veloman-yunkan — COMMENTED** [review]
This is basically an approval. But let's first merge kiwix/libkiwix#700 (upon which this PR depends) and rerun the CI the next day (the only almost-non-issue that I noticed can be used to that end).

File: src/server/kiwix-serve.cpp (hunk 1 above)
Comment (kelson42) [reply src/server/kiwix-serve.cpp:77]: This should be coherent, to me all comments should start with uppercase

Comment (kelson42) [reply src/server/kiwix-serve.cpp:77]: I have fixed it.

**Review by kelson42 — APPROVED** [review]

**Pull request merged**